src/libpst.c

```diff
@@ -1002,17 +1038,25 @@ static int pst_build_id_ptr(pst_file *pf, int64_t offset, int32_t depth, uint64_
         return -1;
     }
     DEBUG_INFO(("Reading index block\n"));
-    if (pst_read_block_size(pf, offset, BLOCK_SIZE, &buf) < BLOCK_SIZE) {
+    if (pst_read_block_size(pf, offset, BLOCK_SIZE, BLOCK_SIZE, &buf) < BLOCK_SIZE) {
         DEBUG_WARN(("Failed to read %i bytes\n", BLOCK_SIZE));
         if (buf) free(buf);
         DEBUG_RET();
         return -1;
     }
     bptr = buf;
-    DEBUG_HEXDUMPC(buf, BLOCK_SIZE, ITEM_SIZE32);
-    item_count = (int32_t)(unsigned)(buf[ITEM_COUNT_OFFSET]);
-    if (item_count > INDEX_COUNT_MAX) {
-        DEBUG_WARN(("Item count %i too large, max is %i\n", item_count, INDEX_COUNT_MAX));
+    DEBUG_HEXDUMPC(buf, BLOCK_SIZE, 0x10);
+    if (pf->do_read64 == 2) {
+        item_count = read_twobyte(buf, ITEM_COUNT_OFFSET);
+        count_max = read_twobyte(buf, MAX_COUNT_OFFSET);
+        LE16_CPU(item_count);
+        LE16_CPU(count_max);
```

Comment:
Oops, just noticed a bug: your read_twobyte macro already converts from little-endian to native, so LE16_CPU will break on big-endian CPUs.